torch/distributions/transforms.py

```diff
@@ -354,19 +355,34 @@ class AffineTransform(Transform):
 
     def __init__(self, loc, scale, event_dim=0, cache_size=0):
         super(AffineTransform, self).__init__(cache_size=cache_size)
-        self.loc, self.scale = broadcast_all(loc, scale)
+        self.loc = loc
+        self.scale = scale
         self.event_dim = event_dim
 
     def __eq__(self, other):
         if not isinstance(other, AffineTransform):
             return False
-        result = self.loc.eq(other.loc).all() and self.scale.eq(other.scale).all()
-        if isinstance(result, Variable):
-            result = result.data.view(-1)[0]
-        return result
+
+        if isinstance(self.loc, numbers.Number) and isinstance(other.loc, numbers.Number):
+            if self.loc != other.loc:
+                return False
+        else:
+            if (self.loc != other.loc).any().item():
+                return False
+
+        if isinstance(self.scale, numbers.Number) and isinstance(other.scale, numbers.Number):
+            if self.scale != other.scale:
+                return False
+        else:
+            if (self.scale != other.scale).any().item():
+                return False
+
+        return True
 
     @property
     def sign(self):
+        if isinstance(self.scale, numbers.Number):
+            return (self.scale > 0) - (self.scale < 0)
```

Comment:
Uh please don’t do arithmetic on Boolean values.

Reply:
Ok, I'll switch to a conditional, but this is the recommended way of writing `cmp(self.scale, 0)` in a Python 2/3 compatible way: https://docs.python.org/3.0/whatsnew/3.0.html#ordering-comparisons